Sources/AblyChat/Message.swift

```diff
@@ -3,16 +3,18 @@ import Foundation
 public typealias MessageHeaders = Headers
 public typealias MessageMetadata = Metadata
 
-public struct Message: Sendable {
+// (CHA-M2) A Message corresponds to a single message in a chat room. This is analogous to a single user-specified message on an Ably channel (NOTE: not a ProtocolMessage).
+// Must conform to Hashable to use some SwiftUI niceties e.g. 'ForEach'.
```

Comment:
weirdly, not true. I replaced this (and other instances where it was needed) with `Identifiable` and still got the same error. 

<img width="798" alt="Screenshot 2024-10-13 at 18 55 38" src="https://github.com/user-attachments/assets/4faab402-5458-4e68-81d7-d9085819b94c">

Reply:
I've updated the comment to highlight the Apple docs seem to be wrong here

Reply:
That error message is telling you that the _ID_ of each message needs to be `Hashable`. It’s failing because you're trying to use a message as its own ID, which doesn't really make sense.

I think that, given there isn't obvious unique identifier to use for a `Message`, it makes sense to use `Hashable`, but the code comment should explain that this is why we're doing it.

Reply:
Updated #10 to mention this too

Reply:
oh I see! I did actually set the ID to the timeserial in the Message struct - just didn't use it here 🤦 

do you think timeserial is a suitable identifier? I don't see how 2 messages in a channel can have the same one?

Reply:
I was wondering about timeserial but I don't really know what it is — is a timeserial a unique identifier for a message?

Reply:
it's essentially a pointer in the channel which is used to know which messages have been seen or not. e.g. the `attachSerial` is a timeserial at the point of attaching to the channel - calling `getPreviousMessages` will get all messages prior to this timeserial. Any subsequent messages will come through the listener and not this call as it's assumed a listener is picking up those messages at that point. 

Every message in turn also has a timeserial, if you were to call `getPreviousMessages` with this timeserial then all messages prior to this message would be returned. 

I'll try use it and see what happens but I don't see how there could be 2 messages with identical timeserials

Reply:
it works - pushed new changes. Marat was also using the timeserial as an ID within the example app